src/prism.c

```diff
@@ -5668,6 +5667,8 @@ context_terminator(pm_context_t context, pm_token_t *token) {
         case PM_CONTEXT_FOR:
         case PM_CONTEXT_ENSURE:
             return token->type == PM_TOKEN_KEYWORD_END;
+        case PM_CONTEXT_FOR_INDEX:
```

Comment:
What's this?

Reply:
When you're parsing the index of a for loop (`for index in collection`) there's slightly different behavior for parsing a multiple target node. So we need to detect that context.